### PR TITLE
fix(shared-research): occasional search crashes

### PR DESCRIPTION
**Pull request by kmaid**

### CODE REVIEW

File: apps/frontend/src/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -34,6 +34,12 @@ const Dashboard: React.FC<Record<string, never>> = () => {
     filters: new Set(),
   });
   usePrefetchCalendars();
+  usePrefetchResearchOutputs({
```
Comment (jeysal) [inline apps/frontend/src/dashboard/Dashboard.tsx:37]: I actually extended `usePrefetchTeams` and some others to have a default config with these values so it doesn't pollute the components that just want some simple prefetching as much. Should use that here as well.

File: apps/frontend/src/shared-research/ResearchOutput.tsx [hunk 2]
```diff
@@ -1,37 +1,25 @@
...
     sharedResearch({}).researchOutput,
   );
-  const { loading, data: researchOutputData } = useResearchOutputById(
-    researchOutputId,
-  );
-
+  const researchOutputData = useResearchOutputById(researchOutputId);
   const backHref = useBackHref() ?? sharedResearch({}).$;
 
-  if (loading) {
```
Comment (jeysal) [inline apps/frontend/src/shared-research/ResearchOutput.tsx:23]: yaaaaay

File: apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx [hunk 3]
```diff
@@ -1,89 +1,92 @@
...
-  })
-    .get('/research-outputs/42')
-    .reply(200, {
-      ...researchOutput,
-      title: 'Proposal title!',
-    });
-  const { getByRole } = await renderComponent('42');
+  mockGetResearchOutput.mockResolvedValue({
+    ...createResearchOutputResponse(),
+    title: 'Proposal title!',
```
Comment (jeysal) [inline apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:74]:
```suggestion
    title: 'PROPOSAL TITLE!',
```
I CAN'T HEAR YOU

Comment (jeysal) [reply apps/frontend/src/shared-research/__tests__/ResearchOutput.test.tsx:74]: (this is a joke btw)

File: apps/frontend/src/shared-research/__tests__/Routes.test.tsx [hunk 4]
```diff
@@ -104,8 +59,10 @@ describe('the library page', () => {
     userEvent.click(checkbox);
     expect(checkbox).toBeChecked();
     await waitFor(() => {
-      const { searchParams } = new URL(req!.path, 'http://api');
-      expect(searchParams.get('filter')).toBe('Proposal');
+      expect(mockGetResearchOutputs).lastCalledWith(
```
Comment (jeysal) [inline apps/frontend/src/shared-research/__tests__/Routes.test.tsx:62]: we should have a lint rule or something to make it clear, but we have used this style exactly zero times in the codebase, so please stick with `toHaveBeenLastCalledWith` :D 
(and below)

File: apps/frontend/src/shared-research/state.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+      researchOutputs.push(researchOutput);
+    }
+    return { total: index.total, items: researchOutputs };
+  },
+  set: (options) => ({ get, set, reset }, researchOutput) => {
+    if (
+      researchOutput === undefined ||
+      researchOutput instanceof DefaultValue
+    ) {
+      const oldEvents = get(researchOutputIndexState(options));
```
Comment (jeysal) [inline apps/frontend/src/shared-research/state.ts:47]: variable name

**Review by jeysal — COMMENTED** [review]
Nice work

File: apps/frontend/src/shared-research/state.ts (hunk 5 above)
Comment (kmaid) [reply apps/frontend/src/shared-research/state.ts:47]: *whistles*

**Review by jeysal — APPROVED** [review]

**Pull request merged**